Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -256,6 +263,40 @@ void AppendPipeline::dispatchDecryptionKey(GstBuffer* buffer)
 }
 #endif
 
+void AppendPipeline::setStartupBufferingComplete(bool complete)
+{
+    LockHolder locker(m_newSampleLock);
+    m_startupBufferingComplete = complete;
+}
+
+void AppendPipeline::flushStartupSamples()
```

Comment:
please, add ASSERT(isMainThread()) or ASSERT(!isMainThread()) to all the new methods to clarify in which thread they're supposed to be running. Also check the {
  braces
}.

Reply:
done